FirebaseAuth/Sources/Swift/AuthProvider/OAuthProvider.swift

```diff
@@ -91,33 +91,104 @@ import Foundation
   /// - Parameter accessToken: The access token associated with the Auth credential be created, if
   /// available.
   /// - Returns: An AuthCredential for the specified provider ID, ID token and access token.
+  @available(swift, introduced: 100.0)
   @objc(credentialWithProviderID:IDToken:accessToken:)
+  public static func credential(withProviderID providerID: NSString,
+                                idToken: NSString,
+                                accessToken: NSString?) -> OAuthCredential {
+    return OAuthCredential(
+      withProviderID: providerID as String,
+      idToken: idToken as String,
+      accessToken: accessToken as String?
+    )
+  }
+
+  @available(
+    *,
+    deprecated,
+    message: "Use `credential(providerID: AuthProviderID, idToken: String, accessToken: String) -> OAuthCredential` instead."
+  )
   public static func credential(withProviderID providerID: String,
                                 idToken: String,
                                 accessToken: String?) -> OAuthCredential {
     return OAuthCredential(withProviderID: providerID, idToken: idToken, accessToken: accessToken)
   }
 
+  /// Creates an `AuthCredential` for the OAuth 2 provider identified by provider ID, ID
+  /// token, and access token.
+  /// - Parameter providerID: The provider ID associated with the Auth credential being created.
+  /// - Parameter idToken: The IDToken associated with the Auth credential being created.
+  /// - Parameter accessToken: The access token associated with the Auth credential be created, if
+  /// available.
+  /// - Returns: An AuthCredential for the specified provider ID, ID token and access token.
+  public static func credential(providerID: AuthProviderID,
+                                idToken: String,
+                                accessToken: String?) -> OAuthCredential {
```

Comment:
@paulb777 , since we're adding a new API here. WDYT about making the accessToken param here default to `nil`? I think it could be a small improvement.

Reply:
+1